src/agent/debugger/src/target.rs

```diff
@@ -567,8 +565,26 @@ impl Target {
         Ok(current_context.get_flags())
     }
 
-    pub fn read_memory(&self, remote_address: LPCVOID, buf: &mut [impl Copy]) -> Result<()> {
-        process::read_memory_array(self.process_handle, remote_address, buf)
+    pub fn read_memory(&mut self, remote_address: LPCVOID, buf: &mut [impl Copy]) -> Result<()> {
+        process::read_memory_array(self.process_handle, remote_address, buf)?;
+
+        // We don't remove breakpoints when processing an event, so it's possible that the
+        // memory we read contains **our** breakpoints instead of the original code.
+        let remote_address = remote_address as u64;
+        let module = self.module_from_address(remote_address);
+        let range = remote_address..=(remote_address + buf.len() as u64);
```

Comment:
Do we want a _non_-inclusive upper bound? I think this can panic if we have a breakpoint set at `remote_address + buf.len()`. If `buf` is already a `&[u8]`, then its max index will be `buf.len() - 1`, but in the loop, we would index into it at `(remote_address + buf.len()) - remote_address == buf.len()`.

Reply:
Yep, good catch.